CHANGELOG.md

```diff
@@ -1,3 +1,14 @@
+## Next
+### Enhancements
+- Added `backoff` feature for reconnecting
+- `send_frame/2` is no longer processed during `opening`, ie. it is queued until `connected` or failure to connect.
+
+### Breaking Changes
+- Connection is always established asynchronously to process init 
+  - `async` option is removed since it's now redundant
+  - `handle_initial_conn_failure` option is removed since it's no longer applicable
```

Comment:
> Hmm, the reason why I removed sync connect is because as far as I know, it's bad practice to handle a potentionally long task in the init stage of a special process

This can definitely be true in the Application supervisor, but sometimes even that is necessary. I don't think you can consider it a bad practice, but it can be annoying to debug. On the other hand, it can also fit really well into the failure as feature mentality.

But there are people who want "footguns" simply because it fits into their application flow easier. That being said, `handle_initial_conn_failure` is a long and slightly convoluted option that is disabled by default for that reason. It was a requested feature, by multiple people iirc.

But my main argument will be that `async` unique startup is a feature. I don't expect `:gen_tcp` to startup then give me an error it can't connect after the incredibly long TCP timeout when I've already done other things with the assumption that I have a valid connection. I wouldn't expect WebSockex to startup without having a WebSocket connection established.